### PR TITLE
fixed VS Code workspace file after v9 upate

### PR DESCRIPTION
**Pull request by gcopoix**
- fixed the make command (use cmake --build instead of make to be independent from cmake generator)
- fixed launched executable filename
- added recommendation for cmake extension

### CODE REVIEW

File: simulator.code-workspace [hunk 1]
```diff
@@ -64,9 +68,9 @@
 				"name": "Debug LVGL demo with gdb",
 				"type": "cppdbg",
 				"request": "launch",
-			        "program": "${workspaceFolder}/build/bin/demo",
+			        "program": "bin/main",
```
Comment (danyyliu) [inline simulator.code-workspace:71]: I have to change this line as below to let it work for me:
"program": "${workspaceFolder}/bin/main",

Comment (danyyliu) [inline simulator.code-workspace:71]: I have to change it as below to let it works for me:
"program": "${workspaceFolder}/bin/main",
"MIMode": "gdb",

Also need to add debug mode in the CmakeLists.txt:
set(CMAKE_BUILD_TYPE Debug)

Comment (danyyliu) [reply simulator.code-workspace:71]: you are right, it works perfect with "cmake::Select Variant"!

**Comment by kisvegabor** [comment]
Thank you for the prompt reply! :heart_eyes: 

I've just tried it out but it doesn't work on my end.
After clicking `Run > Start Debugging` I got this
image
Where I selected the C++ Debugger.

After that this error happened:
image

**Comment by gcopoix** [comment]
Just tested on a clean machine:

1. clone Repo (recursive)
2. checkout feature branch
3. open VSCode by doubleclick on workspace file (or enter `code <path/to/simulator.code-workspace`)
4. let VSCode install recommended plugins (if not already installed)
5. Select `cmake::Configure`, if VSCode asks for kits select `unspecified`, cmake will select the default native gcc)
6. click `Run > Start Debugging` (or press F5)
7. et voilà

May be it makes it more convenient if `cmake.configureOnOpen` is set to `true` in the workspace file, could you please give it a try?
And: It may be a good idea if problems still remain to try with a deleted build folder, especially if basic cmake settings have been changed (or if you have some 'old' build configuration from an earlier version)

**Comment by kisvegabor** [comment]
Getting closer! After removing the build, bin, and the .vscode folders, adding `cmake.configureOnOpen: true` and starting with `simulator.code-workspace` it was configured, on F5 it was compiled, but finally it failed because bin/main is not found:

image

However bin/main exists.

**Comment by kisvegabor** [comment]
`launch.json` looks good too:
```json
"launch": {
		"version": "0.2.0",
		"configurations": [
			{
				"name": "Debug LVGL demo with gdb",
				"type": "cppdbg",
				"request": "launch",
			        "program": "bin/main",
				"args": [],
				"cwd": "${workspaceFolder}",
				"preLaunchTask": "Build",
				"stopAtEntry": false,
			},
		],
	},
```

**Comment by danyyliu** [comment]
I cloned the repo in another directory, started everything from the beginning, and met the same problem:
image

Then I checked the vscode [doc](https://code.visualstudio.com/docs/cpp/launch-json-reference#_configure-vs-codes-debugging-behavior), and it says:
image

So, I am wondering if full path means it should have the env variable workspaceFolder:
"program": "${workspaceFolder}/bin/main"

**Comment by kisvegabor** [comment]
It working well now! Thank you very much! :tada:

**Pull request merged**